### PR TITLE
HOTT-2352: Added table to display additional code and duty

### PR DESCRIPTION
**Pull request by maverickvi**
### Jira link

[HOTT-<TODO>](https://transformuk.atlassian.net/browse/HOTT-2352)

Important: This PR needs to be merged 1 day after https://github.com/trade-tariff/trade-tariff-backend/pull/1010 to allow time for nightly rake tasks to finish

### What?

I have added/removed/altered:

- [ ] Added table to display additional code and duty expression

### Why?

I am doing this because:

- this will help users understand which code belongs to a duty

Screenshot 2023-01-31 at 14 25 58

### CODE REVIEW

File: app/views/commodities/_commodity.html.erb
Comment (willfish) [inline app/views/commodities/_commodity.html.erb]:
```suggestion
              <% tcd.each do |m| %>
```

File: app/webpacker/src/stylesheets/_commodity-tree.scss
Comment (jebw) [inline app/webpacker/src/stylesheets/_commodity-tree.scss]: rather then duplicate govuk sizes, we should defer to them 
- see https://github.com/trade-tariff/trade-tariff-frontend/blob/main/app/webpacker/src/stylesheets/_commodity-codes.scss#L89

And for the margins
- https://design-system.service.gov.uk/styles/spacing/#applying-spacing-in-your-own-css

**Review by jebw — CHANGES_REQUESTED** [review]
Looks good but lets DRY out the css incase GovUK tweak things in the future :raised_hands:

File: app/views/commodities/_commodity.html.erb
Comment (jebw) [inline app/views/commodities/_commodity.html.erb]: This should not require a leading slash - I'm not sure about when in a view (and obviously rendering a partial) but in a controller that actually refers to a specific template file rather a relative one

File: app/views/commodities/_commodity.html.erb
Comment (jebw) [inline app/views/commodities/_commodity.html.erb:27]: This could be `if tcd.many?`, and the second one `tcd.any?`

File: spec/views/commodities/_additional_code_table.html.erb_spec.rb
Comment (jebw) [inline spec/views/commodities/_additional_code_table.html.erb_spec.rb:9]: could use `build_list` here ?

```ruby
let(:measures) { build_list :measure, 2, :vat, :with_additional_code }
```

**Review by jebw — CHANGES_REQUESTED** [review]
Couple of suggestions for coding polish but the only detail I'm keen to see change is to remove the leading slash from the partial name

Not convinced this is actually a problem but seems better to stick to the standard convention

**Review by jebw — APPROVED** [review]
Approved - looks good, can you just remove the spurious `//here` you've added in `_commodity-tree.scss` before merging. Thanks

**Pull request merged**
